### ISSUE TITLE
Oppgradering under Moodle 4.1?

### ISSUE THREAD

**Issue by hgeorgsch** [KQMATH/moodle-mod_capquiz#232]
Installing v0.8.0 på Moodle 4.1.14 the admin reports

```
   Nå har jeg oppdatert CAPQuiz fra versjon 0.6.2 til 0.7.0. Jeg prøvde å
   oppgradere til 0.8.0, men det feilet med

   Exception - syntax error, unexpected '|', expecting '{'

   Jeg vet ikke hva det kan skyldes. Vi har Moodle versjon 4.1.14 på serveren
   nå.
```

Do you know @sebastsg if it is easy to make the changes in v0.8.0 available on Moodle 4.1?  Yes, we cannot spend a lot of time supporting a version reaching end of line, but it is a bit unfortunate since 4.1 was the last LTS until yesterday...

**Comment by sebastsg** [comment]
@hgeorgsch Which PHP version is being used?

**Comment by hgeorgsch** [comment]
PHP 7.4.3 @sebastsg 
Vi går til Moodle 4.5 neste semester med 8.3.

```
Vi kjører PHP 7.4.3 nå, og det kommer til å bli 8.3 på ny server (siden vi i utgangspunktet vil bruke det som standard pakke for Ubuntu 24.04).
Støttes av Moodle 4.5 LTS: https://moodledev.io/general/development/policies/php
```

This does mean that Siebe probably would want to have CAPQuiz working on 4.1 this semester, but if we cannot achieve that rather quickly, we can ignore 4.1 and focus on 4.4 and 4.5

**Comment by sebastsg** [comment]
@hgeorgsch Without looking further into it, it's likely the older PHP version causing this error. I think the time it will take to support PHP 7.4 and Moodle 4.1 is justifiable. If possible however, I'd recommend updating PHP to 8.0 or 8.1, because Moodle 4.1.2+ supports them.

**Comment by sebastsg** [comment]
Released v0.7.1 for Moodle 4.1 and 4.2 (with PHP 7.4 compatibility)
